src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibleBuffer.ts

```diff
@@ -76,66 +73,79 @@ export class AccessibleBufferWidget extends DisposableStore {
 		this._accessibleBuffer.setAttribute('role', 'document');
 		this._accessibleBuffer.ariaRoleDescription = localize('terminal.integrated.accessibleBuffer', 'Terminal buffer');
 		this._accessibleBuffer.classList.add('accessible-buffer');
-		const elt = _xterm.raw.element;
-		if (elt) {
-			elt.insertAdjacentElement('beforebegin', this._accessibleBuffer);
-		}
 		this._editorContainer = document.createElement('div');
 		this._bufferEditor = this._instantiationService.createInstance(CodeEditorWidget, this._editorContainer, editorOptions, codeEditorWidgetOptions);
+		this._accessibleBuffer.replaceChildren(this._editorContainer);
+		this._xtermElement.insertAdjacentElement('beforebegin', this._accessibleBuffer);
+		this._bufferEditor.layout({ width: this._xtermElement.clientWidth, height: this._xtermElement.clientHeight });
+		this.add(this._bufferEditor);
+		this._bufferEditor.onKeyDown((e) => {
+			// tab moves focus mode will prematurely move focus to the next element before
+			// xterm can be focused
+			if (e.keyCode === KeyCode.Escape || e.keyCode === KeyCode.Tab) {
+				e.stopPropagation();
+				e.preventDefault();
+				this._hide();
+			}
+		});
 		this.add(this._configurationService.onDidChangeConfiguration(e => {
 			if (e.affectedKeys.has(TerminalSettingId.FontFamily)) {
 				this._font = _xterm.getFont();
 			}
 		}));
+		this.add(this._xterm.raw.onWriteParsed(async () => {
+			if (this._accessibleBuffer.classList.contains(AccessibleBufferConstants.Active)) {
+				await this._updateEditor(true);
+			}
+		}));
+		this.add(this._bufferEditor.onDidFocusEditorText(() => this._accessibleBuffer.classList.add('active')));
 	}
 
 	private _hide(): void {
-		this._accessibleBuffer.classList.remove('active');
+		this._accessibleBuffer.classList.remove(AccessibleBufferConstants.Active);
+		this._xtermElement.classList.remove(AccessibleBufferConstants.Hide);
 		this._xterm.raw.focus();
 	}
 
-	async show(): Promise<void> {
-		const commandDetection = this._capabilities.get(TerminalCapability.CommandDetection);
-		const fragment = !!commandDetection ? this._getShellIntegrationContent() : this._getAllContent();
-		const model = await this._getTextModel(URI.from({ scheme: AccessibleBufferConstants.Scheme, fragment }));
-		if (model) {
-			this._bufferEditor.setModel(model);
+	private async _updateModel(insertion?: boolean): Promise<void> {
+		let model = this._bufferEditor.getModel();
+		const lineCount = model?.getLineCount() ?? 0;
+		if (insertion && model && lineCount > this._xterm.raw.rows) {
+			const lineNumber = lineCount + 1;
+			model.pushEditOperations(null, [{ range: { startLineNumber: lineNumber, endLineNumber: lineNumber, startColumn: 1, endColumn: 1 }, text: await this._getContent(lineNumber - 1) }], () => []);
+		} else {
+			model = await this._getTextModel(URI.from({ scheme: `${AccessibleBufferConstants.Scheme}-${this._instanceId}`, fragment: await this._getContent() }));
```

Comment:
Should you pass in terminal instance and use `terminalInstance.resource` instead? Not sure why we need a different scheme here

Reply:
without this, one model gets shared between the instances.

Reply:
don't think we need to pass in the terminal instance just to access it's unique identifier